### PR TITLE
Restore sphere orientable state in HPMC

### PR DESCRIPTION
**Pull request by lyrivera**
## Description

Include the orientable flag of spheres as part of the integrator state.

Resolves: #363

### CODE REVIEW

File: hoomd/hpmc/GSDHPMCSchema.h [hunk 1]
```diff
@@ -101,9 +101,13 @@ struct gsd_shape_schema<hpmc::sph_params>: public gsd_schema_hpmc_base
             return 0;
         int retval = 0;
         std::string path = name + "radius";
+        std::string path_o = name + "orientable";
         std::vector<float> data(Ntypes);
+        std::vector<uint32_t> orientableflag(Ntypes);
```
Comment (joaander) [inline hoomd/hpmc/GSDHPMCSchema.h]: Should be `std::vector<uint8_t>`

File: hoomd/hpmc/GSDHPMCSchema.h [hunk 2]
```diff
@@ -115,21 +119,27 @@ struct gsd_shape_schema<hpmc::sph_params>: public gsd_schema_hpmc_base
...
+        std::vector<uint32_t> orientableflag;
         bool state_read = true;
         if(m_exec_conf->isRoot())
             {
             data.resize(Ntypes, 0.0);
+            orientableflag.resize(Ntypes);
             if(!reader->readChunk((void *) &data[0], frame, path.c_str(), Ntypes*gsd_sizeof_type(GSD_TYPE_FLOAT), Ntypes))
                 state_read = false;
+            if(!reader->readChunk((void *) &orientableflag[0], frame, path_o.c_str(), Ntypes*gsd_sizeof_type(GSD_TYPE_UINT32), Ntypes))
+                state_read = false;
```
Comment (joaander) [inline hoomd/hpmc/GSDHPMCSchema.h]: This makes it an error to read state values from existing files.

We need to bump the HOOMD schema version in glotzerlab/gsd#22 to 1,3 and use the following logic here:
```
if (schema <= gsd_make_version(1,2))
    orientable = [0,0,0,0...]
else
    read orientable chunk from file and set state_read appropriately
```

File: hoomd/hpmc/GSDHPMCSchema.h [hunk 3]
```diff
@@ -101,9 +101,13 @@ struct gsd_shape_schema<hpmc::sph_params>: public gsd_schema_hpmc_base
             return 0;
         int retval = 0;
         std::string path = name + "radius";
+        std::string path_o = name + "orientable";
         std::vector<float> data(Ntypes);
+        std::vector<uint32_t> orientableflag(Ntypes);
         std::transform(shape.begin(), shape.end(), data.begin(), [](const hpmc::sph_params& s)->float{return s.radius;});
         retval |= gsd_write_chunk(&handle, path.c_str(), GSD_TYPE_FLOAT, Ntypes, 1, 0, (void *)&data[0]);
+        std::transform(shape.begin(), shape.end(), orientableflag.begin(), [](const hpmc::sph_params& s)->uint32_t{return s.isOriented;});
+        retval |= gsd_write_chunk(&handle, path_o.c_str(), GSD_TYPE_UINT32, Ntypes, 1, 0, (void *)&orientableflag[0]);
```
Comment (joaander) [inline hoomd/hpmc/GSDHPMCSchema.h]: `uint8_t` here as well.

**Comment by lyrivera** [comment]
@joaander The last commit resolves the failing unit tests, per our private discussion.

**Pull request merged**